src/libraries/System.IO.Hashing/tests/XxHash32Tests.007.cs

```diff
@@ -92,6 +92,31 @@ public static IEnumerable<object[]> TestCases
                     "FC23CD03"),
             };
 
+        public static IEnumerable<object[]> LargeTestCases
+        {
+            get
+            {
+                object[] arr = new object[1];
+
+                foreach (LargeTestCase testCase in LargeTestCaseDefinitions)
+                {
+                    arr[0] = testCase;
+                    yield return arr;
+                }
+            }
+        }
+
+        protected static IEnumerable<LargeTestCase> LargeTestCaseDefinitions { get; } =
+            new[]
+            {
+                //https://asecuritysite.com/encryption/xxHash, Example 1
+                new LargeTestCase(
+                    "EEEEE... (10GB)",
+                    (byte)'E',
+                    10L * 1024 * 1024 * 1024, // 10 GB
+                    "1C44F650"),
```

Comment:
Just pedantically double-checking:  We match the reference implementation here?

Just making sure they didn't shut down the accumulator when the length rolled past either 31 or 32 bits.

Reply:
Correct, I ran our inputs against the reference implementation for both 32 and 64 bit variants. The reference implementation appears to simply integer overflow, same as we do.